docs/src/topics/sparse_matrix.md

```diff
@@ -0,0 +1,207 @@
+*nonzero*. In this case the entry `(i, j)` should be included in the sparsity pattern.
+Conversely, if DoFs `i` and `j` *don't* couple, then the computed value will be *zero*. In
+this case the entry `(i, j)` should *not* be included in the sparsity pattern since there is
+no need to allocate memory for entries that will be zero.
+
+The sparsity, i.e. the ratio of zero-entries to the total number of entries, is often *very*
+high and taking advantage of this results in huge savings in terms of memory. For example,
+in a problem with ``10^6`` DoFs there will be a matrix of size ``10^6 \times 10^6``. If all
+``10^{12}`` entries of this matrix had to be stored (0% sparsity) as double precision
+(`Float64`, 8 bytes) it would require 8 TB of memory. If instead the sparsity is 99.9973%
+(which is the case when solving the heat equation on a three dimensional hypercube with
+linear Lagrange interpolation) this would be reduced to 216 MB.
+
+
+### Sparsity pattern example
+
+To give an example, in this one-dimensional heat problem (see the [Heat
+equation](../tutorials/heat_equation.md) tutorial for the weak form) we have 4 nodes with 3
+elements in between. For simplicitly DoF numbers and node numbers are the same but this is
+not true in general since nodes and DoFs can be numbered independently (and in fact are
+numbered independently in Ferrite).
+
+```
+1 ----- 2 ----- 3 ----- 4
+```
+
+Assuming we use linear Lagrange interpolation (the "hat functions") this will give the
+following connections according to the weak form:
+ - Trial function 1 couples with test functions 1 and 2 (entries `(1, 1)` and `(1, 2)`
+   included in the sparsity pattern)
+ - Trial function 2 couples with test functions 1, 2, and 3 (entries `(2, 1)`, `(2, 2)`, and
+   `(2, 3)` included in the sparsity pattern)
+ - Trial function 3 couples with test functions 2, 3, and 4 (entries `(3, 2)`, `(3, 3)`, and
+   `(3, 4)` included in the sparsity pattern)
+ - Trial function 4 couples with test functions 3 and 4 (entries `(4, 3)` and `(4, 4)`
+   included in the sparsity pattern)
+
+The resulting sparsity pattern would look like this:
+
+```
+4×4 SparseArrays.SparseMatrixCSC{Float64, Int64} with 10 stored entries:
+ 0.0  0.0   ⋅    ⋅ 
+ 0.0  0.0  0.0   ⋅ 
+  ⋅   0.0  0.0  0.0
+  ⋅    ⋅   0.0  0.0
+```
+
+Moreover, if the problem is solved with periodic boundary conditions, for example by constraining the
+value on the right side to the value on the left side, there will be additional couplings.
+In the example above, this means that DoF 4 should be equal to DoF 1. Since DoF 4 is
+constrained it has to be eliminated from the system. Existing entries that include DoF 4 are
+`(3, 4)`, `(4, 3)`, and `(4, 4)`. Given the simple constraint in this case we can simply
+replace DoF 4 with DoF 1 in these entries and we end up with entries `(3, 1)`, `(1, 3)`, and
+`(1, 1)`. This results in two new entries: `(3, 1)` and `(1, 3)` (entry `(1, 1)` is already
+included).
```

Comment:
While this is a very nice introductory text, I think it hides the important things below.
Perhaps add at least the example in a collapsible block?